File: client/ruby/lib/ned_client/api/queries_api.rb
require "uri"

module NedClient
  class QueriesApi
    attr_accessor :api_client

    def initialize(api_client = nil)
      @api_client = api_client || Configuration.api_client
    end

    # Get a list of search alerts by type
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :frequency 
    # @option opts [String] :journal 
    # @return [Array<Alert>]
    def get_alerts(opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: QueriesApi#get_alerts ..."
      end
      
      # resource path
      path = "/queries/alerts".sub('{format}','json')

      # query parameters
      query_params = {}
      query_params[:'frequency'] = opts[:'frequency'] if opts[:'frequency']
      query_params[:'journal'] = opts[:'journal'] if opts[:'journal']

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = []
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = []
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = ['basic']
      result = @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Array<Alert>')
      if Configuration.debugging
        Configuration.logger.debug "API called: QueriesApi#get_alerts. Result: #{result.inspect}"
      end
      return result
    end
  end
end

File: client/java/src/main/java/org/plos/ned_client/api/QueriesApi.java
package org.plos.ned_client.api;

import com.sun.jersey.api.client.GenericType;

import org.plos.ned_client.ApiException;
import org.plos.ned_client.ApiClient;
import org.plos.ned_client.Configuration;
import org.plos.ned_client.Pair;

import org.plos.ned_client.model.Alert;

import java.util.*;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-03-03T13:54:47.854-08:00")
public class QueriesApi {
  private ApiClient apiClient;

  public QueriesApi() {
    this(Configuration.getDefaultApiClient());
  }

  public QueriesApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Get a list of search alerts by type
   * 
   * @param frequency 
   * @return List<Alert>
   */
  public List<Alert> getAlerts(String frequency) throws ApiException {
    Object postBody = null;
    
    // create path and map variables
    String path = "/queries/alerts".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    
    queryParams.addAll(apiClient.parameterToPairs("", "frequency", frequency));
    

    

    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "basic" };

    
    GenericType<List<Alert>> returnType = new GenericType<List<Alert>>() {};
    return apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
}
